### PR TITLE
ingress: Improve coverage with unit tests

### PR DESCRIPTION
**Pull request by sayboras**
### Description

This PR is to perform minor clean-up and to improve unit test coverage with test fixtures same as upstream Ingress Conformance tests. The goal is that we don't need to even run Conformance Test to debug any failure/discrepancy.

### Testing

Just to make sure there is no flake in unit tests.

```
$ go test -count=10000 -cover ./operator/pkg/model/...
ok      github.com/cilium/cilium/operator/pkg/model     1.495s  coverage: 37.7% of statements
ok      github.com/cilium/cilium/operator/pkg/model/ingestion   8.095s  coverage: 85.1% of statements
ok      github.com/cilium/cilium/operator/pkg/model/translation 39.045s coverage: 73.6% of statements
ok      github.com/cilium/cilium/operator/pkg/model/translation/gateway-api     17.741s coverage: 84.2% of statements
ok      github.com/cilium/cilium/operator/pkg/model/translation/ingress 6.682s  coverage: 84.0% of statements
```

### CODE REVIEW

File: operator/pkg/model/translation/envoy_virtual_host.go [hunk 1]
```diff
@@ -268,8 +266,7 @@ func getRouteAction(backends []model.Backend) *envoy_config_route_v3.Route_Route
 		Route: &envoy_config_route_v3.RouteAction{
 			ClusterSpecifier: &envoy_config_route_v3.RouteAction_WeightedClusters{
 				WeightedClusters: &envoy_config_route_v3.WeightedCluster{
-					Clusters:    weightedClusters,
-					TotalWeight: wrapperspb.UInt32(uint32(totalWeight)),
```
Comment (mhofstetter) [inline operator/pkg/model/translation/envoy_virtual_host.go:272]: relevant info from the envoy docs (would have been nice to have in the PR description :) )

> Specifies the total weight across all clusters. The sum of all cluster weights must equal this value, if this is greater than 0. **This field is now deprecated, and the client will use the sum of all cluster weights.** It is up to the management server to supply the correct weights.

edit: ah reading the individual commits helps :)

File: operator/pkg/model/translation/ingress/dedicated_ingress_test.go [hunk 2]
```diff
@@ -166,3 +167,60 @@ func Test_getEndpointForIngress(t *testing.T) {
...
+			},
+			want: hostRulesListenersCiliumEnvoyConfig,
+		},
+		{
+			name: "Conformance/PathRules",
+			args: args{
+				m: &model.Model{
+					HTTP: pathRulesListeners,
+				},
+				enforceHTTPs: true,
```
Comment (mhofstetter) [inline operator/pkg/model/translation/ingress/dedicated_ingress_test.go:208]: is the case with `enforceHTTPs=false` covered?

**Review by mhofstetter — APPROVED** [review]

File: operator/pkg/model/translation/ingress/dedicated_ingress_test.go (hunk 2 above)
Comment (sayboras) [reply operator/pkg/model/translation/ingress/dedicated_ingress_test.go:208]: yup, I am planning add more tests to cover such non-default installation.

**Comment by youngnick** [comment]
Beat me to it @mhofstetter, nice work both of you!

**Pull request merged**